pkg/aggregator/aggregator.go

```diff
@@ -897,10 +908,41 @@ func (agg *BufferedAggregator) run() {
 				}
 			}
 			tlmFlush.Add(1, event.eventType, state)
+		case event := <-agg.contLcycleIn:
+			agg.contlcycleDequeueOnce.Do(func() { go agg.dequeueContainerLifecycleEvents() })
+			aggregatorContainerLifecycleEvents.Add(1)
+			agg.handleContainerLifecycleEvent(event)
 		}
 	}
 }
 
+// dequeueContainerLifecycleEvents consumes buffered container lifecycle events.
+// It is blocking, it should run once and in a separate goroutine.
+func (agg *BufferedAggregator) dequeueContainerLifecycleEvents() {
+	for {
+		select {
+		case event := <-agg.contLcycleBuffer:
+			if err := agg.serializer.SendContainerLifecycleEvent(event.msgs, agg.hostname); err != nil {
+				aggregatorContainerLifecycleEventErrors.Add(1)
+				log.Warnf("Error submitting container lifecycle data: %w", err)
+			}
+		case <-agg.stopChan:
```

Comment:
Is this working properly? AFAIK we never close this channel, meaning only one reader will receive a `struct{}` and it could be either your new process loop or the main aggregator process loop.
You may want the aggregator to propagate a message in a dedicated stop channel or similar. I'm not sure we want to close this channel.

Reply:
Nice catch, for my own knowledge why wouldn't  `agg.Stop()` close this channel? I find it very counter-intuitive + it's not ideal if we'll need to add yet another stop chan for this use-case.

Reply:
> Nice catch, for my own knowledge why wouldn't agg.Stop() close this channel? 

Honestly I don't know, it may be the good thing to do after having tried it has no side-effects in current codebase.

Reply:
Based on the current code I believe it's safe to close it. Demux.Stop() is the only caller of agg.Stop() and demux.Stop() destroys the agg instance anyway 
```golang
	if d.aggregator != nil {
		d.aggregator.Stop(flush)
	}
	d.aggregator = nil
```

Reply:
So I tried to rely on closing the channel but it resulted in multiple data races detected in the unit tests - I believe ideally the aggregator shouldn't own its stop chan, potentially the the demux could do it and pass it as argument to the aggregator run method `agg.run(stop)`. But such a refactor is way out of scope here. I choose to have separate stop chan for the contLcycle dequeuer. I hope the whole logic of stopping the aggregator gets improved in the coming demux improvement PRs! 🔥 
Thanks for the help on this @remeh, much appreciated!

Reply:
Ok thanks for the note that's interesting, I guess that's the best for today but indeed, we'll do better in the demux implementation 👍